### PR TITLE
make h option for targz optional, default is set h option

### PR DESCRIPTION
**Pull request by Engerim**

### CODE REVIEW

File: Mage/Task/BuiltIn/Deployment/Strategy/TarGzTask.php [hunk 1]
```diff
@@ -77,7 +77,10 @@ public function run()
             $strategyFlags = '';
         }
 
-        $command = 'tar cfzh' . $strategyFlags . ' ' . $localTarGz . '.tar.gz ' . $excludeCmd . $excludeFromFileCmd . ' -C ' . $this->getConfig()->deployment('from') . ' .';
+        // remove h option only if dump_symlinks is allowed in the release config part
+        $dumpSymlinks = $this->getConfig()->release('dump_symlinks') ? '' : 'h';
```
Comment (eps90) [inline Mage/Task/BuiltIn/Deployment/Strategy/TarGzTask.php]: Keep config naming convention, all multiword config keys are joined with dash (`-`)

**Comment by eps90** [comment]
It would be nice if you would provide some use cases and examples in docs :)

**Comment by Engerim** [comment]
Sure i can provide this, is there a standard, how to write the docs and how to name the doc file?

**Comment by eps90** [comment]
Not yet, but for now it is enough to add some simple example to one (or more) files [in this directory](https://github.com/andres-montanez/Magallanes/tree/master/docs/example-config/.mage/config/environment). 
The coding standards and contribution document is currently in progress, written by me and @jhuet, relating to issue #174 . So be patient, there will be some docs for sure, pretty soon! :)

**Comment by Engerim** [comment]
I added a config with explanation. I  hope it is understandable.

**Comment by eps90** [comment]
Seems reasonable. However I thought about completeing one of existing configs but it's ok :) Let's wait for @andres-montanez opinion, thanks! :)

**Comment by andres-montanez** [comment]
Nice option, thanks!

**Pull request merged**